src/create-constructor-function.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * Creates a constructor function from an ES2015 class, this is a workaround
+ * needed to being able to extend a class from code that's transpiled by Babel.
+ * See https://github.com/babel/babel/pull/8656
+ * @param {*} Type The ES2015 base class used to create the constructor
+ * @param {*} Parent The object where the prototype chain walk to copy over
+ * symbols and static properties to the constructor stops. If not provided,
+ * the chain stops at Object.
+ * @returns {Function} Constructor function than can be safely subclassed from
+ * transpiled code.
+ */
+function createConstructorFunction(Type, Parent) {
+	if (typeof Parent === "undefined") {
```

Comment:
is `Parent` ever intended to be provided?

Reply:
yeah, the idea is/was that when used in `can-stache-element` for example you'd prevent it from going to `HTMLElement` ... in `can-observable-array` you could pass `Array` and get it to stop there?